docs/Local-Development.md

```diff
@@ -0,0 +1,164 @@
+# Local development
+
+1. [First steps | Local installation](#first-steps)
+1. [Setup testing env](#setup-testing-env)
+1. [Test commands locally](#test-commands-locally)
+1. [Cleanup](#cleanup)
+
+## First Steps
+
+**Get an OpsLevel API token for a dev environment.** Do not use your prod environment since this involves mutating lots of services in OpsLevel.
+
+Install these:
+
+1. [minikube](https://minikube.sigs.k8s.io/docs/start/)  - this is a suggestion, you can also use **your own dev cluster if you have one.** Other options include [kind](https://kind.sigs.k8s.io/) or [k3s](https://k3s.io/).
+1. [task](https://taskfile.dev/) - this will install dependencies for you.
+1. [OpsLevel CLI](https://github.com/OpsLevel/cli) - makes things way easier.
+
+```
+brew install minikube
+brew install task
+brew install opslevel/tap/cli
+```
+
+Then run:
+
+```
+task setup
+cd src                       # from now on this tutorial is going to assume you're in the src/ directory.
+go build .
+./kubectl-opslevel           # you can also use `go run main.go` or set an alias like `alias ko='./kubectl-opslevel'`
```

Comment:
Added -o flag, but I do not want to touch taskfiles in this PR for now since the scope is very big.

By the way the reason why these changes are included in this PR is because I set the wrong base on https://github.com/OpsLevel/kubectl-opslevel/pull/241. This stuff was supposed to be on main after the signal handling was merged to main.